webui/react/src/components/kit/Icon.tsx

```diff
@@ -117,11 +204,28 @@ const Icon: React.FC<Props> = ({
 }: Props) => {
   const classes = [css.base];
 
+  const rowIcon = React.useMemo(() => {
+    if (size === 'small') return <RowIconSmall />;
+    if (size === 'large') return <RowIconLarge />;
+    if (size === 'big') return <RowIconExtraLarge />;
+
+    return <RowIconMedium />; // returns medium by default
+  }, [size]);
+  const svgIcon = React.useMemo(() => {
```

Comment:
sugg: at this point I'd suggest making a switch statement like the following:
```
const icon = useMemo(() => {
switch (name) {
    case "columns":
        return <ColumnsIcon />;
...
    default:
        return <span aria-label={title} className={classes.join(' ')} />;
}
})

Also, nit: I would import `useMemo` directly rather than doing `React.useMemo`